java-checks-testkit/src/test/java/org/sonar/java/checks/verifier/internal/CheckVerifierTestUtils.java

```diff
@@ -189,6 +189,22 @@ public void scanFile(JavaFileScannerContext context) {
     }
   }
 
+  @Rule(key = "VerifyRootProjectWorkingDirectory")
+  protected static final class VerifyRootProjectWorkingDirectory implements JavaFileScanner {
+    private final String rootProjectWorkingDirectory;
+
+    public VerifyRootProjectWorkingDirectory(String rootProjectWorkingDirectory){
+      this.rootProjectWorkingDirectory = rootProjectWorkingDirectory;
+    }
+
+    @Override
+    public void scanFile(JavaFileScannerContext context) {
+      if(!rootProjectWorkingDirectory.equals(context.getRootProjectWorkingDirectory().getPath())){
+        throw new RuntimeException("Expected rootProjectWorkingDirectory does not match context.getRootProjectWorkingDirectory");
```

Comment:
Looking at the test in [InternalCheckVerifierTest](https://github.com/SonarSource/sonar-java/blob/3d14b7f92bcb1c33a7f278af371a2edc82ba655c/java-checks-testkit/src/test/java/org/sonar/java/checks/verifier/internal/InternalCheckVerifierTest.java), I don't understand how we ever go through this line

Reply:
Is this check necessary?

Reply:
Let's move this check to `context_return_good_root_working_directory` as an anonymous class and replace the `if` statement with an assertion.